### PR TITLE
Add theta-SL-pg2 config for EAM Antarctica RRM

### PR DESCRIPTION
**Pull request by wlin7**
theta-SL-pg2 configuration and supporting files are added for grid
ne0np4_antarcticax4v1.pg2_r0125_ne0np4_antarcticax4v1.pg2
with ocean mask oRRS15to5.

[BFB]

### CODE REVIEW

**Comment by wlin7** [comment]
Hi @ambrad and @tangq , this PR mirrors #3549 for northamericax4v1.pg2 with theta-l dycore.
It is based on the ne0np4_antarcticax4v1 you created.

I will do some simulation tests with this grid soon.

**Comment by wlin7** [comment]
I am using  map_antarcticax4v1pg2_to_r0125_mono map for ATM2LND_SMAP and FMAP, because check_map gives strong warning on bilin map. BTW, northamericax4v1.pg2 uses bilin map for ATM2LND_SMAP. I also did a check_map on it. It gives strong warning as well. Would that be a concern?

**Comment by ambrad** [comment]
Some of chk_map's warnings apply only for global-to-global maps but fire for global-to-nonglobal maps and vice versa. For example, for map_northamericax4v1pg2_to_WC14to60E2r3_bilin.20200924.nc, the important output is
```
Grid A size n_a: 57816 // Number of columns/sources
mask_a 0's, 1's: 0, 57816
mask_a min, max: 1, 1
area_a sum/4*pi: 0.9999999999999952 = 1.0-4.8e-15 // Perfect is 1.0 for global Grid A
area_a min, max: 1.3883814992116144e-05, 7.9828290235106039e-04
Column (source cell) indices utilized min, max: 1, 57816
Ignored source cells (empty columns): 18565

Consistency metrics (row-sums of weights) and errors
Perfect metrics for global Grid A are avg = min = max = 1.0, mbs = rms = sdn = 0.0:
frac_b avg: 1.0000000000000000 = 1.0-0.0e+00 // Mean
frac_b min: 0.9999999999999998 = 1.0-2.2e-16 // Minimum in grid B cell [81,-1.01058,+2.12364]
frac_b max: 1.0000000000000002 = 1.0+2.2e-16 // Maximum in grid B cell [41,-0.602306,+2.75371]
frac_b mbs: 0.0000000000000000 =     1.6e-17 // Mean absolute bias from 1.0
frac_b rms: 0.0000000000000001 =     5.1e-17 // RMS relative to 1.0
frac_b sdn: 0.0000000000000001 =     5.1e-17 // Standard deviation
```
Generally, for atm->ocn, the consensus is we need `bilin` (monotone, nonconservative interpolation) rather than `mono` for the state map to provide higher-order state fields.

**Comment by mt5555** [comment]
@wlin7 
at the bottom of this page:

https://acme-climate.atlassian.net/wiki/spaces/ED/pages/178848194/Transition+to+TempestRemap+for+Atmosphere+grids

there is a summary of what is expected from ncks --chk_map for non global grids and/or non-conservative state maps.

**Comment by tangq** [comment]
@wlin7 , I would keep the mapping algorithms the same as the North America RRM unless your tests fail or show unphysical behavior.

File: components/eam/bld/namelist_files/namelist_defaults_cam.xml [hunk 1]
```diff
@@ -259,6 +259,7 @@
 <bnd_topo hgrid="ne0np4_northamericax4v1"  >atm/cam/topo/USGS_northamericax4v1_12xdel2_consistentSGH_191023.nc</bnd_topo>
 <bnd_topo hgrid="ne0np4_northamericax4v1" npg="2">atm/cam/topo/USGS_northamericax4v1pg2_12xdel2_consistentSGH_20020209.nc</bnd_topo>
 <bnd_topo hgrid="ne0np4_antarcticax4v1"  >atm/cam/topo/USGS-gtopo30_antarcticax4v1_12xdel2_consistentSGH_191120.nc</bnd_topo>
+<bnd_topo hgrid="ne0np4_antarcticax4v1"  npg="2">atm/cam/topo/USGS_gtopo30_aantarcticax4v1pg2_12xdel2_consistentSGH_20200925.nc</bnd_topo>
```
Comment (ambrad) [inline components/eam/bld/namelist_files/namelist_defaults_cam.xml:262]: `aantarctica` is misspelled here.

**Review by tangq — APPROVED** [review]
Looks good to me.

File: components/eam/bld/namelist_files/namelist_defaults_cam.xml (hunk 1 above)
Comment (maltrud) [reply components/eam/bld/namelist_files/namelist_defaults_cam.xml:262]: antarctica is also misspelled on the title of the PR, but in a different way.

**Comment by wlin7** [comment]
Thank you @ambrad , @mt5555 , @tangq for the tips on check_map report. For the bilin map, the weight and consistency metrics are good. The conservation metrics give alarming numbers, for both northamericax4v1pg2 and antarcticax4v1pg2 to r0125 grid. 

```
Conservation metrics (column-sums of area_b-weighted weights normalized by area_a) and errors
Perfect metrics for global Grid B are avg = min = max = 1.0, mbs = rms = sdn = 0.0:
frac_a avg: 40625853724646.4453125000000000 = 1.0+4.1e+13 // Mean
frac_a min: 32217949109201.0742187500000000 = 1.0+3.2e+13 // Minimum in grid A cell [25242,+41.2746,+216.791]
frac_a max: 60542001567187.8906250000000000 = 1.0+6.1e+13 // Maximum in grid A cell [20174,+31.9598,+224.218]
frac_a mbs: 40625853724645.3906250000000000 =     4.1e+13 // Mean absolute bias from 1.0
frac_a rms: 40663314467199.0859375000000000 =     4.1e+13 // RMS relative to 1.0
frac_a sdn: 1745051653290.9309082031250000 =     1.7e+12 // Standard deviation

```
Can it be ignored, even though it is not a conservative map? The conservation metrics for map_northamericax4v1pg2_to_WC14to60E2r3_bilin.20200924.nc are much smaller numbers, though check_map also gives warnings.

**Comment by ambrad** [comment]
The conservation numbers are meaningless and OK in this case. If you point me to where I can find the bilin map, I can make an image like this one, which I've made already for mono:

map_antarcticax4v1pg2_to_r0125_mono 20200925 nc A

Sorry about the inconvenient longitude wrapping of the r0125 grid at the poles. Images are (top left) test field on atm, (top right) test field on lnd, (bottom left) atm->lnd remapped on lnd, (bottom right) log10 error on lnd.

**Comment by wlin7** [comment]
> The conservation numbers are meaningless and OK in this case. 

That is a relieve to me, @ambrad , because the magnitudes between different maps differ by so much in terms of this metric.

The bilin map I referred to is available on following path
```
anvil: /home/ac.wlin/ACME/remap/antarcticax4v1.pg2/map_antarcticax4v1pg2_to_r0125_bilin.20200925.nc
compy: /qfs/people/linw288/E3SM/remap/antarcticax4v1.pg2/map_antarcticax4v1pg2_to_r0125_bilin.20200925.nc

```
Thanks.

**Comment by ambrad** [comment]
Great. Looks good:
map_antarcticax4v1pg2_to_r0125_bilin 20200925 nc A

**Comment by wlin7** [comment]
Thanks a lot @ambrad . I was scared away by those conservation metric numbers. I am going to replace the SMAP with the bilin map.

**Comment by ambrad** [comment]
Great. In the case of the interpolation bilin map, I think the area values are meaningless, which is why the conservation metrics in `ncks --chk_map` in turn are meaningless.

File: components/eam/bld/namelist_files/namelist_defaults_cam.xml (hunk 1 above)
Comment (wlin7) [reply components/eam/bld/namelist_files/namelist_defaults_cam.xml]: Thanks for the sharp eyes, @ambrad , @maltrud . Corrected. Double checked the file in inputdata server is good.

**Comment by mt5555** [comment]
I think the reason the conservation metrics are meaningless for bilin maps is because ESMF leaves out the cell area for this algorithm.  I tried to document this on that page linked above.

**Review by ambrad — APPROVED** [review]
I'm approving based on visual inspection. I recommend you do a run before merging to master, which I think you are indeed planning. I also recommend you grep the log files for nc filenames and check that they match what you expect.

**Comment by wlin7** [comment]
> I'm approving based on visual inspection. I recommend you do a run before merging to master, which I think you are indeed planning. I also recommend you grep the log files for nc filenames and check that they match what you expect.

Yes, I did. Running well on cori. Checked the related files in atm_in and seq_map.rc as well. Thanks again.

**Comment by wlin7** [comment]
Hi @golaz , @rljacob , can this PR go in during code freeze? It has no effect on WC; will be used for cryo simulations.

**Comment by rljacob** [comment]
Yes.  No impact on WC.

**Comment by wlin7** [comment]
Thanks, @rljacob . Merged to next.

**Comment by wlin7** [comment]
Merged to master.

**Pull request merged**